### ISSUE TITLE
Error with time.strptime() while parsing timestamps with AM/PM strings 

### ISSUE THREAD

**Issue by pinno** [spyder-ide/spyder#10669]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports.
Yes, issue 498 is similar (https://github.com/spyder-ide/spyder/issues/498) but it is also outdated (nearly 5 years ago).
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [x] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
Yes, the code works correctly with Jupyter qtconsole and also with IPython and standard Python consoles.
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``. I dont' want to lose my preferences, sorry.
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

I cannot parse timestamp strings containing the AM/PM characters. The error appeared initially with pandas: `data.index = pd.to_datetime(data['Timestamp'], format='%d/%m/%Y %I:%M:%S %p')`, then I tried by simply using `time.strptime()`, which also failed as you can see below.

### What steps reproduce the problem?

1. `from time import strptime`
2. `strptime('19/06/2019 07:49:53 PM', '%d/%m/%Y %I:%M:%S %p')`

### What is the expected output? What do you see instead?

The expected output is:
```
time.struct_time(tm_year=2019, tm_mon=6, tm_mday=19, tm_hour=19, tm_min=49, tm_sec=53, tm_wday=2, tm_yday=170, tm_isdst=-1)
```

Instead I see the following traceback:

### Paste Traceback/Error Below (if applicable)

```python-traceback

Traceback (most recent call last):

  File "<ipython-input-4-56063b868d6c>", line 2, in <module>
    strptime('19/06/2019 07:49:53 PM', '%d/%m/%Y %I:%M:%S %p')

  File "/home/work/anaconda3/lib/python3.7/_strptime.py", line 571, in _strptime_time
    tt = _strptime(data_string, format)[0]

  File "/home/work/anaconda3/lib/python3.7/_strptime.py", line 362, in _strptime
    data_string[found.end():])

ValueError: unconverted data remains: PM

```

These two lines run correctly with Python and IPython consoles. I think this issue is NOT related with issue 498, since the locales are the same in both Spyder and IPython:

IPython:
```
In [1]: import locale                                                                                                                      
In [2]: locale.getlocale()                                                                                                                 
Out[2]: ('en_US', 'UTF-8')
```

Spyder:
```
In [1]: import locale
In [2]: locale.getlocale()
Out[2]: ('en_US', 'UTF-8')
```

## Versions

Python 3.7.3 64-bit | Qt 5.9.6 | PyQt5 5.9.2 | Linux 4.4.0-166-generic 
* Spyder version: 3.3.6
* Python version: 3.7.3 64-bit
* Qt version: 5.9.6
* PyQt version: 5.9.2
* Operating System name/version: Linux 4.4.0-166-generic (Ubuntu 16.04.6 LTS)

### Dependencies

```
IPython >=4.0     :  7.9.0 (OK)
cython >=0.21     :  0.29.13 (OK)
jedi >=0.9.0      :  0.15.1 (OK)
matplotlib >=2.0.0:  3.1.1 (OK)
nbconvert >=4.0   :  5.6.1 (OK)
numpy >=1.7       :  1.17.3 (OK)
pandas >=0.13.1   :  0.25.2 (OK)
psutil >=0.3      :  5.6.3 (OK)
pycodestyle >=2.3 :  2.5.0 (OK)
pyflakes >=0.6.0  :  2.1.1 (OK)
pygments >=2.0    :  2.4.2 (OK)
pylint >=0.25     :  2.4.3 (OK)
qtconsole >=4.2.0 :  4.5.5 (OK)
rope >=0.9.4      :  0.14.0 (OK)
sphinx >=0.6.6    :  2.2.1 (OK)
sympy >=0.7.3     :  1.4 (OK)
```

**Comment by ccordoba12** [comment]
@steff456, please try to reproduce this on Linux.

**Comment by rotemgb** [comment]
It works fine for me on Linux 5.3.0-20-generic, Pop_Os 19.10, in both spyder 3.3.6 and 4.0.0.dev0.

**Comment by ccordoba12** [comment]
@rotemgb, thanks a lot for the verification!

@pinno, I'm sorry but I doubt this is a problem with Spyder because there's nothing pointing to our code in the traceback you posted above. My suggestion is to create a new conda env from scratch and try things there or reinstall Anaconda.

**Comment by pinno** [comment]
Hi, thanks for your time. In the meanwhile yesterday I momentarily stopped using Spyder and I used IPython for developing my software, and I've experienced a similar error by applying the conversion of timestamps to datetimes with Pandas: `data.index = pd.to_datetime(data['Timestamp'], format='%d/%m/%Y %I:%M:%S %p')`
The run on the first dataset executes correctly, while the run on the second dataset (within the same script) fails with the same error as above. I solved by aggregating the datasets and running the conversion to datetime only once within the script.
So, it's not an issue with Spyder, even if the behaviour of this bug is different with respect to e.g. IPython.
Next step is creating a new conda environment and check again for this weird behaviour.
